library/blivet.py

```diff
@@ -1265,6 +1276,50 @@ def _manage_encryption(self, members):
 
         return managed_members
 
+    def _manage_members(self):
+        """ Schedule actions as needed to configure this pool's members. """
+        if not self._device:
+            return
+
+        add_disks = [d for d in self._disks if d not in self._device.ancestors]
+        remove_disks = [pv for pv in self._device.pvs if not any(d in pv.ancestors for d in self._disks)]
+
+        if not (add_disks or remove_disks):
+            return
+
+        if self._is_raid:
+            raise BlivetAnsibleError("managing pool members is not supported with RAID")
+
+        for disk in add_disks:
+            member = self._create_one_member(disk)
+            member = self._manage_one_encryption(member)
+
+            try:
+                ac = ActionAddMember(self._device, member)
+                self._blivet.devicetree.actions.add(ac)
+            except Exception as e:
+                raise BlivetAnsibleError("failed to add disk '%s' to pool '%s': %s" % (disk.name,
+                                                                                       self._pool['name'],
+                                                                                       str(e)))
+
+        for disk in remove_disks:
```

Comment:
Question: Should we consider removing the PV *unsafe*? Currently the code will remove the PV from the VG even with the safe mode on. This isn't a destructive operation, but I am also removing the PV from the disk after that and that might be potentially dangerous. And it also might not be user friendly to remove the PV just because user specified incomplete `disks` by accident.

Reply:
>  The storage_safe_mode description in the README states that "When true (the default), an error will occur instead of automatically removing existing devices and/or formatting.", and it seems that this operation would fall under "removing existing devices".

Thanks, I'll change the code to require safe mode to be turned off for this operation.

> Does the role fail if the user tries to remove a PV from a VG and the PV has data on it?

If there is enough free space in the VG it will run `pvmove` to move the LV to an empty PV but it will return error if there isn't enough free space.